cmd/fleet/serve.go

```diff
@@ -722,6 +722,21 @@ func cronVulnerabilities(
 			sentry.CaptureException(err)
 		}
 
+		if err := ds.CleanUpUnusedSoftware(ctx); err != nil {
```

Comment:
@mna

I've moved the cleanup performed in `CalculateHostsPerSoftware` to a new `CleanUpUnusedSoftware` (to make it more explicit and also for the principle of least surprise). Let me know if it makes sense.

Reply:
Yeah, as long as it is called immediately after `CalculateHostsPerSoftware`, that's fine (and I agree with your explanation). It must be done immediately after to properly identify unused software right after the counts got updated.

Reply:
After some thinking I ended up reverting the change (due to the dependency between the two operations, calculation and cleanup it's less error prone to keep them united).